app/questionnaire/questionnaire_schema.py

```diff
@@ -852,14 +852,19 @@ def _get_rules_section_dependencies(
             if not isinstance(rule, Mapping):
                 continue
 
-            answer_id = None
+            answer_id_list: list = []
+            identifier: str = rule.get("identifier", "")
```

Comment:
Maybe we shouldn't use a fallback option if the specified key doesn't exist at this point (checked by validator)?
```suggestion
            identifier: str = rule.get("identifier")
```
otherwise we should probably check if identifier exists in the elif statements.